lib/light-service/action.rb

```diff
@@ -67,7 +82,9 @@ def rolled_back
       private
 
       def create_action_context(context)
-        return context if context.is_a? LightService::Context
+        usable_defaults(context).each do |ctx_key, default|
```

Comment:
`map` wouldnt be appropriate here as we're not returning a new enumerable, we're mutating the candidate context blob to apply defaults if they're required.

`usable_defaults(ctx)` takes the list of missing expected keys, and returns a trimmed hash of available defaults that contains only the missing keys that have a default.

It then loops through that hash and applies the default to the context.

Can you walk me through how this would work with a map?